core/src/main/scala/implicits.scala

```diff
@@ -70,10 +68,52 @@ object implicits {
 
   }
 
-  implicit def listenableFutureHandler(implicit s: Session): ListenableFutureHandler =
-    new ListenableFutureHandler
+  class ClusterAPIHandler[M[_]](implicit H: ListenableFuture[?] ~> M, E: MonadError[M, Throwable])
+      extends ClusterAPI.Handler[ClusterAPIOps[M, ?]] {
+
+    def connect: ClusterAPIOps[M, Session] = Kleisli(c => H(c.connectAsync()))
+
+    def connectKeyspace(keyspace: String): ClusterAPIOps[M, Session] =
+      Kleisli(c => H(c.connectAsync(keyspace)))
+
+    def close: ClusterAPIOps[M, Unit] = closeFuture2unit[M, Cluster](_.closeAsync())
+
+    def configuration: ClusterAPIOps[M, Configuration] =
+      Kleisli(c => E.catchNonFatal(c.getConfiguration))
+
+    def metadata: ClusterAPIOps[M, Metadata] =
+      Kleisli(c => E.catchNonFatal(c.getMetadata))
+
+    def metrics: ClusterAPIOps[M, Metrics] =
+      Kleisli(c => E.catchNonFatal(c.getMetrics))
+
+  }
+
+  private[this] def closeFuture2unit[M[_], A](f: A => CloseFuture)(
```

Comment:
Maybe this method could be placed inside the handler class, since it seems to be an internal utility for this specific handler. In that case, you don't need type parameters and the `H` NT.